components/org.wso2.carbon.identity.api.server.branding.preference.management/org.wso2.carbon.identity.api.server.branding.preference.management.v1/src/main/java/org/wso2/carbon/identity/api/server/branding/preference/management/v1/core/BrandingPreferenceManagementService.java

```diff
@@ -0,0 +1,423 @@
+            if (RESOURCE_NOT_EXISTS_ERROR_CODE.equals(e.getErrorCode())) {
+                return false;
+            }
+            throw handleConfigurationMgtException
+                    (e, ERROR_CODE_ERROR_CHECKING_BRANDING_PREFERENCE_EXISTS, getTenantDomainFromContext());
+        }
+        if (resource == null) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Generate and return resource name of the default branding of the particular tenant.
+     *
+     * @return resource name of the default branding resource.
+     */
+    private String getDefaultResourceName() {
+
+        int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+        String resourceName = tenantId + RESOURCE_NAME_SEPARATOR + DEFAULT_LOCALE;
+        return resourceName;
+    }
+
+    /**
+     * Build a resource object from Branding Preference Model.
+     *
+     * @param model       Branding Preference Model.
+     * @param inputStream Branding Preference file stream.
+     * @return Resource object.
+     */
+    private Resource buildResourceFromBrandingPreference(BrandingPreferenceModel model, InputStream inputStream) {
+
+        /**
+         * Currently this API provides the support to only configure tenant wise branding preference for 'en-US' locale.
+         * So always use resource name as default resource name.
+         */
+        String resourceName = getDefaultResourceName();
+        Resource resource = new Resource();
+        resource.setResourceName(resourceName);
+        // Set file.
+        ResourceFile file = new ResourceFile();
+        file.setName(resourceName);
+        file.setInputStream(inputStream);
+        List<ResourceFile> resourceFiles = new ArrayList<>();
+        resourceFiles.add(file);
+        resource.setFiles(resourceFiles);
+        return resource;
+    }
+
+    /**
+     * Build a Branding Preference Model from branding preference file stream.
+     *
+     * @param inputStream Branding Preference file stream.
+     * @return Branding Preference Model.
+     */
+    private BrandingPreferenceModel buildBrandingPreferenceFromResource(InputStream inputStream)
+            throws IOException {
+
+        String preferencesJSON = IOUtils.toString(inputStream, StandardCharsets.UTF_8.name());
+        if (!BrandingPreferenceUtils.isValidJSONString(preferencesJSON)) {
+            throw handleException
+                    (Response.Status.INTERNAL_SERVER_ERROR, ERROR_CODE_ERROR_BUILDING_RESPONSE_EXCEPTION, null);
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
+        Object preference = mapper.readValue(preferencesJSON, Object.class);
+        BrandingPreferenceModel brandingPreferenceModel = new BrandingPreferenceModel();
+        brandingPreferenceModel.setPreference(preference);
+        brandingPreferenceModel.setType(BrandingPreferenceModel.TypeEnum.valueOf(ORGANIZATION_TYPE));
```

Comment:
Pass the type and local to this method and reuse the variables.

Reply:
Addressed with https://github.com/wso2/identity-api-server/pull/332/commits/3a13da85f5dc2357d5c9babfa715a99b18bae6c6